### PR TITLE
Formdata and console output updates

### PR DESCRIPTION
**Pull request by weerd**
This PR makes an important update to not force `FormData` passed as part of a `post`, `put` or `patch` to be interpreted as a JSON string and thus mangle the data resulting in nothing getting passed through to the ending URL. To get around this before, we'd resort to _not_ using `Gateway` and doing everything directly through `window.fetch()`, which defeats the purpose of having a package like this!

It also does some updates to make the console output a little nicer and cleans up the code by moving the customized console outputs over to the `helpers.js` file as separate methods that can be called.

New console log output:
image

New console error output:
image

Also, on the side of the consumer using this API Client, if Gateway throws a `GatewayError`, the consumer can `catch()` this error and within it, there should be a `response` which contains the parsed JSON of the original response, so you get the endpoints expressed error data:

Example, using `console.log(error.response)` within a `catch()`:
image

### CODE REVIEW

File: src/RestApiClient.js [hunk 1]
```diff
@@ -14,7 +15,7 @@ class RestApiClient {
         'Content-Type': 'application/json',
         'Accept': 'application/json',
       },
-      credentials: overrides.credentials || 'same-origin',
+      credentials: 'same-origin',
```
Comment (weerd) [inline src/RestApiClient.js]: We didn't need to set this with `overrides.credentials`. If there is an override for this, it gets replaced when the objects are `merge()`ed in the follow up statement.

File: src/RestApiClient.js [hunk 2]
```diff
@@ -163,23 +178,13 @@ class RestApiClient {
...
+    gatewayLog(method, url, options);
 
     return window.fetch(url, options)
       .then(this.checkStatus)
       .then(this.parseJson)
-      .catch((data) => {
-        console.log(data);
-        console.error('um, there was an error! handle that shizzzzz!');
+      .catch((error) => {
+        gatewayError(error);
```
Comment (weerd) [inline src/RestApiClient.js:169]: Aims to provide more information than just `um, there was an error! handle that shizzzzz!` 🎉

File: src/RestApiClient.js [hunk 3]
```diff
@@ -162,26 +163,49 @@ class RestApiClient {
...
+
+        throw error;
       });
   }
+
+  /**
+   * Set the configuration options for the client.
+   *
+   * @see https://github.com/github/fetch/issues/505#issuecomment-293064470
```
Comment (weerd) [inline src/RestApiClient.js:180]: This is was driving me nuts! 😵

File: src/RestApiClient.js [hunk 4]
```diff
@@ -162,26 +163,49 @@ class RestApiClient {
...
     return window.fetch(url, options)
       .then(this.checkStatus)
       .then(this.parseJson)
-      .catch((data) => {
-        console.log(data);
-        console.error('um, there was an error! handle that shizzzzz!');
+      .catch((error) => {
+        gatewayError(error);
+
+        throw error;
```
Comment (weerd) [inline src/RestApiClient.js]: This also drove me nuts, but less-so than the `multipart/form-data` content type issue. 🤕

File: src/RestApiClient.js (hunk 3 above)
Comment (mendelB) [reply src/RestApiClient.js:180]: Whoah, interesting one.

**Review by mendelB — APPROVED** [review]
(WIth the disclaimer that I'm quite unfamiliar with this codebase,) these changes look good to me! 👍

**Comment by mendelB** [comment]
Welp, didn't see the new changes 🤦‍♂️

File: src/RestApiClient.js [hunk 5]
```diff
@@ -37,11 +39,12 @@ class RestApiClient {
...
     } else {
-      const error = new Error(response.statusText);
+      throw new GatewayError(response, response.statusText);
+      // const error = new Error(response.statusText);
 
-      error.response = response;
+      // error.response = response;
 
-      throw error;
+      // throw error;
```
Comment (mendelB) [inline src/RestApiClient.js]: Just a couple o comment 👻's here

**Review by mendelB — APPROVED** [review]
One small thing, but still looks sharp!

File: src/RestApiClient.js (hunk 5 above)
Comment (weerd) [reply src/RestApiClient.js]: dang. I am sleepy 💤

Comment (weerd) [reply src/RestApiClient.js]: good eye! 👁

**Comment by weerd** [comment]
image

*unused code.... I'm so 💤

**Comment by weerd** [comment]
Forgot to give a quick shoutout to @jeder28 for helping me with the Promises logic in the caught error, since I'm so bad at them 💃

**Pull request merged**